### PR TITLE
[add] functionnality to process recordset by slice

### PR DESCRIPTION
**Pull request by damdam-s**

### CODE REVIEW

**Comment by damdam-s** [comment]
@leemannd the case is you have a bunch of records and the method you want to trigger can last for a while.

With this, anthem gives you a feedback about the processing like

```
    Processing slice 0/73
    Processing slice 1/73
    Processing slice 2/73
    Processing slice 3/73
    Processing slice 4/73
    Processing slice 5/73
    Processing slice 6/73
    Processing slice 7/73
    Processing slice 8/73
    Processing slice 9/73
    Processing slice 10/73
    Processing slice 11/73
    Processing slice 12/73
    Processing slice 13/73
    Processing slice 14/73
    Processing slice 15/73
    Processing slice ...
```

File: anthem/lyrics/records.py [hunk 1]
```diff
@@ -59,3 +59,16 @@ def switch_company(ctx, company):
     ctx.env.user.company_id = safe_record(ctx, company)
     yield ctx
     ctx.env.user.company_id = current_company
+
+
+def process_by_slice(ctx, records, method, step=1000):
+    total_length = len(records)
```
Comment (grindtildeath) [inline anthem/lyrics/records.py]: Please add a docstring

**Comment by leemannd** [comment]
AFAIR, the importer we are using is also giving informations

**Comment by guewen** [comment]
@damdam-s What was your use case?

**Comment by leemannd** [comment]
Closing as the author left the company and we never dig further the question.

**Pull request closed without merging**